### PR TITLE
Add the error function (erf)

### PR DESCRIPTION
**Pull request by muupan**
This PR adds the error function (erf).

CPU forward computation requires scipy. Please check if it is the correct way to use scipy in Chainer.

### CODE REVIEW

**Comment by muupan** [comment]
I'll try supporting environments without scipy by using math.erf instead.

File: chainer/functions/__init__.py [hunk 1]
```diff
@@ -180,6 +180,8 @@
 from chainer.functions.math.det import batch_det  # NOQA
 from chainer.functions.math.det import BatchDet  # NOQA
 from chainer.functions.math.det import det  # NOQA
+from chainer.functions.math.erf import erf  # NOQA
+from chainer.functions.math.erf import Erf  # NOQA
```
Comment (beam2d) [inline chainer/functions/__init__.py]: We have decided not to expose newly added `Function`/`FunctionNode` classes under `chainer.functions`, so remove this line.

File: tests/chainer_tests/functions_tests/math_tests/test_erf.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    def check_label(self, op, expected):
+        self.assertEqual(op().label, expected)
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(3, 2), ()],
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+}))
+class TestErf(UnaryFunctionsTestBase):
```
Comment (beam2d) [inline tests/chainer_tests/functions_tests/math_tests/test_erf.py]: Could you consider using `unary_math_function_unittest` utility to generate test cases? See the usage here: https://docs.chainer.org/en/stable/reference/generated/chainer.testing.unary_math_function_unittest.html#chainer.testing.unary_math_function_unittest

File: chainer/functions/math/erf.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+def erf(x):
+    """Elementwise error function.
+
+    .. note::
+       Forward computation in CPU can be slow if
+       `SciPy <https://www.scipy.org/>`_ is not available.
+
+    Args:
+        x (~chainer.Variable): Input variable.
```
Comment (beam2d) [inline chainer/functions/math/erf.py]: Add `numpy.ndarray` and `cupy.ndarray` for allowed argument types.

**Comment by muupan** [comment]
Sorry for my late response. I fixed all the issues you mentioned.

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**